src/mon/MonClient.h

```diff
@@ -139,8 +144,65 @@ class MonConnection {
   MessageRef pending_tell_command;
 
   AuthRegistry *auth_registry;
+  std::string mon_name;
 };
 
+struct MonClientQuorum {
+private:
+  CephContext *cct;
+public:
+  std::map<entity_addrvec_t, std::pair<version_t, std::set<int>>> quorums;
```

Comment:
Do we need to store addresses? Isn't the intention something like "who told us about the quorum-at-epoch?". If so, perhaps ranks would be fine.

Reply:
i took the address since the rank can be changed

Reply:
but not in the same epoch?